specification/09-coveio_abi.adoc

```diff
@@ -74,99 +183,358 @@ struct sbiret sbi_covh_reclaim_tvm_interface_region(unsigned long tvm_id,
+[#sbi_covg_get_device_link]
+===== Function: CoVE Guest Get Device Link (FID TBD)
 [source, C]
 -----
-struct sbiret sbi_covg_get_device_link(unsigned long device_id);
+struct sbiret sbi_covg_get_device_link(unsigned long device_if_id);
 -----
 
-[#sbi_covg_get_device_certificate()]
-===== Device Identity Certificate (FID TBD)
+Gets the status of the link between the physical device hosting `device_if_id`
+and the TVM. This covers both the SPDM and IDE links.
+
+Returns the a link status bitmap through `sbiret.value`.
+
+[source, C]
+-------
+// A secure SPDM Session is established.
+#define SECURED_SPDM (1 << 0)
+
+// A PCIe IDE link is established.
+#define PCIE_IDE (1 << 1)
+-------
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_covg_get_device_link]
+.CoVE Guest Get Device Link
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#sbi_covg_get_device_certificate]
+===== Function: CoVE Guest Get Device Certificate (FID TBD)
 [source, C]
 -----
-struct sbiret sbi_covg_get_device_certificate(unsigned long device_id);
+struct sbiret sbi_covg_get_device_certificate(unsigned long device_if_id,
+                                                       unsigned long cert_addr_out,
+                                                       unsigned long cert_size);
 -----
 
-[#sbi_covg_get_device_measurements()]
-===== Device Measurements (FID TBD)
+Get the certificate chain for the physical device hosting `device_if_id`.
+
+The TVM calls this function in order to verify the authenticity of the physical
+device. A TVM must not accept a device interface without doing that verification
+first.
+
+`cert_addr_out` must be page aligned.
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_covg_get_device_certificate]
+.CoVE Guest Get Device Certificate
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+[#sbi_covg_get_device_measurements]
+===== Function: CoVE Guest Get Device Measurements (FID TBD)
 [source, C]
 -----
-struct sbiret sbi_covg_get_device_measurements(unsigned long device_id);
+struct sbiret sbi_covg_get_device_measurements(unsigned long device_if_id,
+                                                        unsigned long msmt_addr_out,
+                                                        unsigned long msmt_size);
 -----
 
-[#sbi_covg_get_device_spdm_attrs()]
-===== Device SPDM Session Attributes (FID TBD)
+Gets the measurements of the the physical device hosting `device_if_id`.
+
+The TSM returns the measurements in the form of concatenated <<SPDM>> blocks,
```

Comment:
We cannot only return Measurement Block, because the attestation may need opaque data or even full transcript.

Can we return the full measurement transcript here?

Reply:
Then would it make sense to also passe the nonce from the `sbi_covg_get_device_measurements` caller?

Reply:
I think so, but I think it could be optional.

Reply:
I fixed that section, let me know if it looks good to you.